src/main/java/tc/oc/pgm/rotation/FixedPGMMapOrderManager.java

```diff
@@ -0,0 +1,225 @@
+  /** When a {@link PGMMap} is manually set next, it overrides the rotation order * */
+  private PGMMap overriderMap;
+
+  public FixedPGMMapOrderManager(MatchManager matchManager, Logger logger, File rotationsFile) {
+    this.matchManager = matchManager;
+    this.rotationsFile = rotationsFile;
+    this.logger = logger;
+
+    if (!rotationsFile.exists()) {
+      try {
+        FileUtils.copyInputStreamToFile(PGM.get().getResource("rotations.yml"), rotationsFile);
+      } catch (IOException e) {
+        logger.log(Level.SEVERE, "Failed to create the rotations.yml file", e);
+      }
+    }
+
+    loadRotations();
+  }
+
+  private void loadRotations() {
+    FileConfiguration rotationsFileConfiguration =
+        YamlConfiguration.loadConfiguration(rotationsFile);
+
+    rotationsFileConfiguration
+        .getKeys(false)
+        .forEach(
+            key ->
+                rotations.add(
+                    new FixedPGMMapOrder(
+                        rotationsFileConfiguration.getConfigurationSection(key), key)));
+    rotations.forEach(FixedPGMMapOrder::load);
+
+    rotations.forEach(
+        rotation -> {
+          if (!rotation.isEnabled()) rotations.remove(rotation);
+        });
+
+    setActiveRotation(getRotationByName("default"));
+  }
+
+  private void setActiveRotation(FixedPGMMapOrder activeRotation) {
+    this.activeRotation = activeRotation;
+  }
+
+  public FixedPGMMapOrder getActiveRotation() {
+    return activeRotation;
+  }
+
+  void setEvaluatingPlayerCount(boolean evaluatingPlayerCount) {
+    this.isEvaluatingPlayerCount = evaluatingPlayerCount;
+  }
+
+  public boolean isEvaluatingPlayerCount() {
+    return isEvaluatingPlayerCount;
+  }
+
+  public List<FixedPGMMapOrder> getRotations() {
+    return rotations;
+  }
+
+  public void saveCurrentPosition() {
+    FileConfiguration rotationFileConfiguration =
+        YamlConfiguration.loadConfiguration(rotationsFile);
+
+    rotationFileConfiguration.set(
+        activeRotation.getName() + ".next_map",
+        activeRotation.getMapInPosition(activeRotation.getPosition() + 1).getName());
+
+    try {
+      rotationFileConfiguration.save(rotationsFile);
+    } catch (IOException e) {
+      logger.log(
+          Level.SEVERE, "Could not save current rotation's position for future reference", e);
+    }
+  }
+
+  public void recalculateActiveRotation() {
+    int activePlayers = getActivePlayers();
+    List<Integer> playerCounts = new ArrayList<>(rotations.size());
+
+    rotations.forEach(
+        rotation -> {
+          if (rotation.isEnabled()) {
+            playerCounts.add(rotation.getPlayers());
+          }
+        });
+
+    Collections.sort(playerCounts);
+    FixedPGMMapOrder newRotation = null;
+
+    int count = 0;
+    while (count < playerCounts.size()) {
+      newRotation = getRotationByPlayerCount(playerCounts.get(count));
+
+      if (playerCounts.get(count) >= activePlayers) {
+        break;
+      }
+
+      count++;
+    }
```

Comment:
Seems like a weird mix of list, foreach, sort, and while, that could be simplified to a single stream
```java
FixedPGMMapOrder newRotation = rotations.stream()
   .filter(FixedPGMMapOrder::isEnabled)
    .map(FixedPGMMapOrder::getPlayers)
    .sorted()
    .filter(playerCount -> playerCount >= activePlayers)
    .map(this::getRotationByPlayerCount)
    .findFirst().orElse(null);
```